packages/ember-data/lib/system/store.js

```diff
@@ -1397,10 +1397,11 @@ Store = Service.extend({
   */
 
   _modelForMixin: function(key) {
-    var mixin = this.container.resolve('mixin:' + key);
+    var registry = this.container._registry;
```

Comment:
Not super happy with grabbing `_registry` here. Accessing private members like this has caused problems for ED in the past. Is there a better way to do this @dgeb / @stefanpenner?